repo/blob/b2/b2_storage.go

```diff
@@ -149,13 +158,20 @@ func (s *b2Storage) PutBlob(ctx context.Context, id blob.ID, data blob.Bytes, op
 		r = http.NoBody
 	}
 
-	_, err := s.bucket.UploadFile(fileName, nil, r)
+	fi, err := s.bucket.UploadFile(fileName, timestampmeta.ToMap(opts.SetModTime, timeMapKey), r)
+	if err != nil {
+		return translateError(err)
+	}
 
-	return translateError(err)
-}
+	if opts.GetModTime != nil {
+		if opts.SetModTime.IsZero() {
+			*opts.GetModTime = time.Unix(0, fi.UploadTimestamp*1e6)
+		} else {
+			*opts.GetModTime = opts.SetModTime
```

Comment:
nit; suggestion to use the time from the response either way

Reply:
done